src/browser/js/DesktopConnector.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-2016 Yuya Ochiai
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import EventEmitter from 'events';
+
+/**
+ * Class to handle interfacing with the loaded webapp
+ */
+export default class DesktopConnector extends EventEmitter {
+  /**
+   * Sends data to the webapp for the provided event
+   * @param {string} event
+   * @param  {...any} data
+   */
+  send(event, ...data) {
+    if (window.desktopBridge && window.desktopBridge.ready) {
```

Comment:
Similarly, instead of using the global instance of `DesktopBridge` here, couldn't the `DesktopBridge` just pass a reference to itself when creating the `DesktopConnector` like `this.registerDesktopConnector(new DesktopConnector(this));`? Then the `DesktopConnector` would be able to refer back to the bridge.